### PR TITLE
Get the reset logic working on logical timer 2 (Physical 2)

### PR DESCRIPTION
**Pull request by stieg**
Due to the hardware wiring bug we have no way to have the slave
controller reset the counter when an interrupt comes in.  Thus
the only way we can reset the counter is... not.  This patch
adds code so that we account for roll over, count the ticks per
overflow period, and wisely choose when to reset the output vs
not.  This way we emulate the effects of the slave reset without
having one.

### CODE REVIEW

File: stm32_base/hal/timer_stm32/timer_device_stm32.c [hunk 1]
```diff
@@ -377,26 +380,30 @@ void TIM1_BRK_TIM9_IRQHandler(void)
...
         uint32_t current = TIM_GetCapture4(TIM2);
         if (last < current) {
-            uint32_t delta = current - last;
-            timer2_period = delta;
+                timer2_period = current - last;
+        } else {
+                timer2_period = TIMER_PERIOD - last + current;
         }
         last = current;
+        ++sigs;
```
Comment (brentpicasso) [inline stm32_base/hal/timer_stm32/timer_device_stm32.c]: what does sigs stand for? can we make it more explicit?

Comment (stieg) [inline stm32_base/hal/timer_stm32/timer_device_stm32.c]: Stands for signals.  It counts the number of times an interrupt occurs on a CC register

Comment (brentpicasso) [inline stm32_base/hal/timer_stm32/timer_device_stm32.c]: Ok. the name was confusing.  How about trigger_count, or irq_count?

**Comment by brentpicasso** [comment]
Other than variable name, looks good AFAICT. :)  Testing will tease out the rest.

**Comment by stieg** [comment]
Variables updated.  Good to go @brentpicasso ?

**Comment by brentpicasso** [comment]
ship it!

On Thu, Feb 18, 2016 at 7:22 AM, Andrew Stiegmann <notifications@github.com>
wrote:

> Variables updated. Good to go @brentpicasso
> <https://github.com/brentpicasso> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/432#issuecomment-185771529>
> .
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*

**Pull request merged**